options.go

```diff
@@ -0,0 +1,417 @@
+package httpsling
+
+import (
+	"encoding/base64"
+	"net/http"
+	"net/url"
+	"strings"
+	"unicode"
+
+	"github.com/ansel1/merry"
+	goquery "github.com/google/go-querystring/query"
+
+	"github.com/theopenlane/httpsling/httpclient"
+)
+
+// Option applies some setting to a Requester object
+type Option interface {
+	Apply(*Requester) error
+}
+
+// OptionFunc adapts a function to the Option interface
+type OptionFunc func(*Requester) error
+
+// Apply implements Option
+func (f OptionFunc) Apply(r *Requester) error {
+	return f(r)
+}
+
+// With clones the Requester, then applies the options to the clone
+func (r *Requester) With(opts ...Option) (*Requester, error) {
+	r2 := r.Clone()
+
+	err := r2.Apply(opts...)
+	if err != nil {
+		return nil, err
+	}
+
+	return r2, nil
+}
+
+// MustWith clones the Requester, then applies the options to the clone
+func (r *Requester) MustWith(opts ...Option) *Requester {
+	if r2, err := r.With(opts...); err != nil {
+		panic(err)
+	} else {
+		return r2
+	}
+}
```

Comment:
nit: you shouldn't need the `else` 
```suggestion
func (r *Requester) MustWith(opts ...Option) *Requester {
	r2, err := r.With(opts...)
	if err != nil {
		panic(err)
	}

	return r2
}
```